lib/graphql/schema.rb

```diff
@@ -1030,13 +1030,21 @@ def references_to(to_type = nil, from: nil)
             own_refs + inherited_refs
           end
         else
-          find_inherited_value(:references_to, EMPTY_HASH).merge(@own_references_to)
+          # `@own_references_to` can be quite large for big schemas,
+          # and generally speaking, we won't inherit any values.
+          # So optimize the most common case -- don't create a duplicate Hash.
+          inherited_value = find_inherited_value(:references_to, EMPTY_HASH)
+          if inherited_value.any?
+            inherited_value.merge(@own_references_to)
+          else
+            @own_references_to
+          end
         end
       end
 
       def type_from_ast(ast_node, context: nil)
-        type_map = context ? context.warden.types : self.types
```

Comment:
Calling `.types` can be slow because it merges some big hashes. This was the _only_ use at runtime, so I found a way to replace it!